src/app/frontend/podlist/podcard_component.js

```diff
@@ -82,7 +83,59 @@ export class PodCardController {
    * @export
    */
   isFailed() {
-    return this.pod.podStatus.podPhase === 'Failed' || this.hasWarnings();
+    if (this.pod.podStatus.podPhase === 'Failed') {
+      return true;
+    }
+
+    if (this.pod.podStatus.podPhase === 'Succeeded') {
+        return false;
+    }
+
+    // This will return true if the pod has events that indicate warnings while in a Pending state.
+    if (this.hasWarnings() && this.pod.podStatus.podPhase !== 'Running') {
+        return true;
+    }
+
+    return this.isInitialized() && !this.isReady();
+  }
+
+  /**
+   * Checks if pod status is ready. i.e. it's running and passess readiness checks.
+   * @return {boolean}
+   * @export
+   */
+  isReady() {
+    let ready = false;
```

Comment:
Should this default to true? I.e., are we sure that pod condition of type "Ready" will always be there?